root/user/ReportUser.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import noop from 'lodash/noop';
+import * as React from 'react';
+
+import {addColon, l, N_l} from '../static/scripts/common/i18n';
+import UserAccountLayout from '../components/UserAccountLayout';
+import FormRow from '../components/FormRow';
+import FormRowCheckbox from '../components/FormRowCheckbox';
+import FormRowSelect from '../components/FormRowSelect';
+import FormRowTextArea from '../components/FormRowTextArea';
+import FormSubmit from '../components/FormSubmit';
+import {withCatalystContext} from '../context';
+
+type ReportReasonT =
+  | 'enforcing_guidelines'
+  | 'ignoring_guidelines'
+  | 'other'
+  | 'spam'
+  | 'unresponsiveness'
+  | 'voting'
+  ;
+
+type Props = {|
+  +$c: CatalystContextT,
+  +form: FormT<{|
+    +message: FieldT<string>,
+    +reason: FieldT<ReportReasonT>,
+    +reveal_address: FieldT<boolean>,
+  |}>,
+  +user: EditorT,
+|};
+
+const reportReasonOptions = {
+  grouped: false,
+  options: [
+    {
+      label: N_l('Editor is spamming'),
+      value: 'spam',
+    },
+    {
+      label: N_l('Editor is unresponsive to edit notes'),
+      value: 'unresponsiveness',
+    },
+    {
+      label: N_l('Editor intentionally ignores accepted guidelines'),
+      value: 'ignoring_guidelines',
+    },
+    {
+      label: N_l('Editor is overzealous in enforcing guidelines as rules'),
+      value: 'enforcing_guidelines',
+    },
+    {
+      label: N_l('Editor engages in overzealous or abusive yes/no voting'),
+      value: 'voting',
+    },
+    {
+      label: N_l('Editor has violated some other part of our Code of Conduct'),
+      value: 'other',
+    },
+  ],
+};
+
+const ReportUser = ({
+  $c,
+  form,
+  user,
+}: Props) => {
+  form.field.reveal_address.value = true;
```

Comment:
A default value should go in lib/MusicBrainz/Server/Form/User/Report.pm instead (you can pass a `default` key to the `reveal_address` field). This would reset it to true even if they previously unchecked it but the page reloaded due to an unrelated error.